### PR TITLE
Add `RENDERING_INFO` parameters to GL Compatibility renderer

### PR DESCRIPTION
**Pull request by clayjohn**
This is part of completion of the GL Compatibility renderer. Users can now see rendering info in the monitors and in the 3D viewport.

This also fixes RENDERING_INFO_TOTAL_PRIMITIVES_IN_FRAME for the RD renderers as it was incorrectly reporting vertex/index count at times.

<details>
  <summary>background to RENDERING_INFO_TOTAL_PRIMITIVES_IN_FRAME changes</summary>
The rendering info parameters were modified in https://github.com/godotengine/godot/pull/50096. After some discussion, and right before merging, reduz switched from reporting vertices to reporting primitives. Unfortunately, some references to vertices were not cleaned up, so in parts of the codebase it still referred to vertices, despite reporting primitive count. Additionally, the code accidentally reported vertex count when using non-LODed meshes (despite calculating the proper vertex count). It was also wrongly ignoring instance count in this case.

These mistakes led to a lot of confusion, which eventually led to https://github.com/godotengine/godot/pull/59558 which changed the label in the editor to wrongly report "Primitive Indices". I have updated the label again to say "Primitives (points, lines, triangles)" to make it clear what is being reported. 
 
</details>

One notable difference between OpenGL and Vulkan is that in the OpenGL renderer "Video Mem" is just the combination of texture mem and buffer mem as we don't have a memory allocator to track total memory usage the way we do with the RenderingDevice. 

Finally, this PR introduces a warning when the Utilities class is destructed and we haven't freed all texture or buffer memory. Hopefully this will help us stay on top of memory leaks. 

The warning is currently reporting leaked texture memory, so before we merge this PR I need to figure out if this is actual leaked memory, or just a failure to track when memory is being freed. 

Screenshot from 2023-05-26 17-59-02
image

CC @BastiaanOlij

### CODE REVIEW

**Comment by akien-mga** [comment]
> These mistakes led to a lot of confusion, which eventually led to #59558 which changed the label in the editor to wrongly report "Primitive Indices". I have updated the label again to say "Primitives (points, lines, triangles)" to make it clear what is being reported.

I agree with the rationale, but the new name is quite long, which leads to a wide overlay for this info, unbalanced compared to the other lines. Languages like German would make it even worse. Maybe we can find a shorter compromise?

**Comment by BastiaanOlij** [comment]
Looks good @clayjohn 

On the subject of the label, I would just keep it to Primitives and maybe explain it somewhere in the documentation and maybe leave a comment in the code for the next person who thinks its more limited then it is.

**Comment by clayjohn** [comment]
> > These mistakes led to a lot of confusion, which eventually led to #59558 which changed the label in the editor to wrongly report "Primitive Indices". I have updated the label again to say "Primitives (points, lines, triangles)" to make it clear what is being reported.
> 
> I agree with the rationale, but the new name is quite long, which leads to a wide overlay for this info, unbalanced compared to the other lines. Languages like German would make it even worse. Maybe we can find a shorter compromise?

How about "Primitives (TIWAGOS)"?

**Comment by clayjohn** [comment]
Updated to return to just "Primitives". The documentation clarifies that "primitives" means lines, points, or triangles

**Review by BastiaanOlij — APPROVED** [review]
Looks good @clayjohn , my only remark, and I realise that I'm way too late to the party with this, is that I would have created wrapper functions in utilities for each gl function that allocated/deallocates data. 

So say `glBufferData` becomes `GLES3::Utilities::get_singleton()->glBufferData`, or possibly something a little more appropriately named as we may want to change the parameters away from it's GL counterpart.

In this function we would bind the buffer, push in the data, and then modify our allocated memory variables. We could also maintain a map for each ID to store how much memory is allocated against that ID so when we can use that when an existing buffer is overwritten with new data, or when the buffer is freed.

It would become much easier to ensure we modify the variables correctly, and provide visual queues in code when we're still calling the GL functions directly.

Also we could add functionality to name entries in our map and provide much more detailed information on leaked information, similar to what we're doing in rendering device.

File: drivers/gles3/rasterizer_scene_gles3.cpp [hunk 1]
```diff
@@ -2744,6 +2783,12 @@ RasterizerSceneGLES3::~RasterizerSceneGLES3() {
 	memdelete_arr(scene_state.omni_light_sort);
 	memdelete_arr(scene_state.spot_light_sort);
 
+	uint32_t light_buffer_size = GLES3::Config::get_singleton()->max_renderable_lights * sizeof(LightData);
+	uint32_t directional_light_buffer_size = MAX_DIRECTIONAL_LIGHTS * sizeof(DirectionalLightData);
+	GLES3::Utilities::get_singleton()->buffer_mem_cache -= light_buffer_size;
```
[the remark this answers is not in the document]
Comment (clayjohn) [reply drivers/gles3/rasterizer_scene_gles3.cpp]: I changed how this works, so this is no longer an issue

**Comment by clayjohn** [comment]
Okay, I updated to use helper functions in the Utilities class to allocate and free so we get tracking automatically. I also added hashmaps to track resource name/size, so we can identify specific instances that leak during dev builds

**Comment by clayjohn** [comment]
@akien-mga Is there a way to suppress codespell? Its emitting a false positive on lines of code that aren't touched by this PR

**Review by BastiaanOlij — APPROVED** [review]
Looks really good

**Comment by akien-mga** [comment]
> @akien-mga Is there a way to suppress codespell? Its emitting a false positive on lines of code that aren't touched by this PR

I already got it narrowed down to only modified files, but we can't get it finer than that. I'll fix the issue in master.

**Comment by akien-mga** [comment]
Thanks!

**Comment by clayjohn** [comment]
> > @akien-mga Is there a way to suppress codespell? Its emitting a false positive on lines of code that aren't touched by this PR
> 
> I already got it narrowed down to only modified files, but we can't get it finer than that. I'll fix the issue in master.
> 
> _Edit:_ Fixed by #78316. I rebased this PR to fix CI.

Thank you!

**Pull request merged**